tests/installation/system_role.pm

```diff
@@ -51,16 +50,20 @@ sub assert_system_role {
     # Still initializing the system at this point, can take some time
     # Asserting screen with preselected role
     # Proper default role assertion will be addressed in poo#37504
-    assert_screen 'system-role-default-system', 180;
-    my $system_role = get_var('SYSTEM_ROLE', 'default');
-    if (get_var('SYSTEM_ROLE') && !check_var('SYSTEM_ROLE', 'default')) {
-        change_system_role($system_role);
+    # Product might or might not have default selected
+    if (is_opensuse) {
+        assert_screen('before-role-selection', 180);
+        change_system_role(get_var('DESKTOP'));
```

Comment:
Sorry for the confusion, but as I've mentioned in the last comment we still need SYSTEM_ROLE variable, as we still have lxde, xfce and minimalx, which doesn't have defined mapping to SYSTEM_ROLE. So I would keep previous version with adding either check_var('SYSTEM_ROLE', 'default') or check_var('SYSTEM_ROLE', 'gnome'), because most of the scenarios we have on o3 having gnome as graphical system. Or alternatively set in in main.pm for obvious cases, like these 3 which we address here. I just believe it will be confusing when we will select gnome or server role, when we have lxde or xfce.

Reply:
What about in line 56:  `change_system_role(get_var('SYSTEM_ROLE'), get_var('DESKTOP'));`?